### PR TITLE
Rename CollectionRetrievalResult field 'batch' to 'transaction_list'

### PR DESCRIPTION
**Pull request by arun-koshy**
Fixes [Issue#529](https://github.com/MystenLabs/narwhal/issues/529)

### CODE REVIEW

**Review by akichidis — COMMENTED** [review]
I would recommend to change the response structure on the `CollectionRetrievalResult` so we align it with the inner concept of batches & transactions. I think mixing the concept of Batches with transactions might be a bit more confusing. To recap our Block / Collection is comprised of multiple batches and each batch of multiple transactions. So a Collection is something like:

```
Collection
|---> Batch 
|         |---> transaction
|         |---> transaction
|         |...........
|----> Batch
|         |----> transaction
|         |----> transaction
......
```

So this is what I would suggest to forward with our API as well. Since in our response we are returning multiple collections then we probably want to return a vector of collections. So a `Vec<Collection>` . @arun-koshy thoughts?

File: examples/protos/narwhal.proto [hunk 1]
```diff
@@ -58,7 +58,7 @@ message ValidatorData {
 
 message CollectionRetrievalResult {
     oneof retrieval_result {
-        BatchMessage batch = 1;
+        BatchMessage transaction_list = 1;
```
Comment (huitseeker) [inline examples/protos/narwhal.proto:61]: The problem here is not so much the name 'batch', it's the reuse of "BatchMessage". I.e. this still makes folks believe we are reusing the Batch concept to convey this transaction list. I think it's better to use a different type.

File: primary/tests/integration_tests_validator_api.rs [hunk 2]
```diff
@@ -925,15 +928,18 @@ async fn test_get_collections_with_missing_certificates() {
...
     );
 
     for result in actual_result {
         match result.retrieval_result.unwrap() {
-            RetrievalResult::Batch(batch) => {
-                let id: BatchDigest = batch.id.unwrap().into();
-                let result_batch: Batch = batch.transactions.unwrap().into();
+            RetrievalResult::TransactionList(transaction_list) => {
+                let id: BatchDigest = transaction_list.id.unwrap().into();
+                let result_batch: Batch = transaction_list.transactions.unwrap().into();
```
Comment (huitseeker) [inline primary/tests/integration_tests_validator_api.rs:942]: This use of `BatchDigest` and `Batch` is the one that is illegitimate: the digest in here is a `CertificateDigest`, and the transaction list itself should have its own type.

**Review by huitseeker — COMMENTED** [review]
I think there are two topics:
1. conceptually, do we expose the internal batching structure when we retrieve a collection? That's certainly something we can revisit, but if so, perhaps not in PR comments.
2. the actual types we expose at the moment in a `CollectionRetrievalResult` should have nothing to do with a `BatchDigest` (they are not comparable to any `BatchDigest` in the system) or with a `BatchMessage` (the are the concatenation of several batches, and thus not a batch).

I've left comments in this direction in the PR.

**Comment by arun-koshy** [comment]
Just to clarify `BlockWaiter` returns 

```
pub struct GetBlocksResponse {
    pub blocks: Vec<BlockResult<GetBlockResponse>>,
}
```

```
pub struct GetBlockResponse {
    pub id: CertificateDigest,
    pub batches: Vec<BatchMessage>,
}
```

```
pub struct BatchMessage {
    // TODO: revisit including the id here [see #188]
    pub id: BatchDigest,
    pub transactions: Batch,
}
```

```
pub struct Batch(pub Vec<Transaction>);
```

```
pub type Transaction = Vec<u8>;
```

Currently we are just placing the `batches` in `GetBlockResponse` into a proto as follows

```
message CollectionRetrievalResult {
    oneof retrieval_result {
        BatchMessage batch = 1;
        CollectionError error = 2;
    }
}
```

```
message BatchMessage {
    BatchDigest id = 1;
    Batch transactions = 2;
}
```

```
message Batch {
    repeated Transaction transaction = 1;
}
```

```
message Transaction {
    bytes transaction = 1;
}
```

But if I am understanding @huitseeker and @akichidis correctly we want to wrap the batches in the original `Collection` and remove BatchMessage to simplify the type tree i.e.

```
message CollectionRetrievalResult {
    oneof retrieval_result {
        Collection collection = 1;
        CollectionError error = 2;
    }
}
```

```
message Collection {
    CertificateDigest id = 1;
    repeated Batch batches = 2;
}
```

```
message Batch {
    BatchDigest id = 1;
    repeated Transaction transactions = 2;
}
```

```
message Transaction {
    bytes transaction = 1;
}
```

File: primary/tests/integration_tests_validator_api.rs (hunk 2 above)
Comment (arun-koshy) [reply primary/tests/integration_tests_validator_api.rs:942]: I actually threw away the `CertificateDigest` in the original implementation and only returned the `BatchDigest` and `Batch`. Explained further below.

**Comment by akichidis** [comment]
> I think there are two topics:
> 
> 1. conceptually, do we expose the internal batching structure when we retrieve a collection? That's certainly something we can revisit, but if so, perhaps not in PR comments.
> 2. the actual types we expose at the moment in a `CollectionRetrievalResult` should have nothing to do with a `BatchDigest` (they are not comparable to any `BatchDigest` in the system) or with a `BatchMessage` (the are the concatenation of several batches, and thus not a batch).
> 
> I've left comments in this direction in the PR.

@huitseeker regarding topic (1) yes I believe we should revisit this endpoint. More specifically in the aspects of:
* returned items / structure
* errors
* naming

Thus I don't recommend doing any renaming/changes and redesign it. I agree, let's not use the PR, we can setup a discussion for this.

**Review by akichidis — APPROVED** [review]
That looks great @arun-koshy 💯  thanks so much!

**Pull request merged**